spec/tapioca/dsl/compilers/protobuf_spec.rb

```diff
@@ -161,26 +152,19 @@ def clear_cart_item_index; end
             end
 
             it "generates methods in RBI files for classes with Protobuf with enum field" do
-              add_ruby_file("protobuf.rb", <<~RUBY)
-                require 'google/protobuf/timestamp_pb'
-                require 'google/protobuf/wrappers_pb'
-
-                Google::Protobuf::DescriptorPool.generated_pool.build do
-                  add_file("cart.proto", :syntax => :proto3) do
-                    add_message "MyCart" do
-                      optional :value_type, :enum, 1, "VALUE_TYPE"
-                    end
-                    add_enum "VALUE_TYPE" do
-                      value :NULL, 0
-                      value :FIXED_AMOUNT, 1
-                      value :PERCENTAGE, 2
-                    end
-                  end
-                end
+              add_proto_file("cart", <<~PROTO)
+                syntax = "proto3";
 
-                Cart = Google::Protobuf::DescriptorPool.generated_pool.lookup("MyCart").msgclass
-                Cart::VALUE_TYPE = Google::Protobuf::DescriptorPool.generated_pool.lookup("VALUE_TYPE").enummodule
-              RUBY
+                message Cart {
+                  optional VALUE_TYPE value_type = 1;
+                }
+
+                enum VALUE_TYPE {
+                  NULL = 0;
+                  FIXED_AMOUNT = 1;
+                  PERCENTAGE = 2;
+                }
```

Comment:
What happens if we move this definition inside the `message Cart` definition? Can we replicate the `Cart` namespacing for the enum that way?

Reply:
Good catch, moving it inside does keep the namespacing.